plugins/woocommerce/includes/integrations/maxmind-geolocation/class-wc-integration-maxmind-geolocation.php

```diff
@@ -168,7 +168,10 @@ public function validate_license_key_field( $key, $value ) {
 	public function update_database( $new_database_path = null ) {
 		// Allow us to easily interact with the filesystem.
 		require_once ABSPATH . 'wp-admin/includes/file.php';
-		WP_Filesystem();
+		if ( ! WP_Filesystem() ) {
+			wc_get_logger()->notice( __( 'Failed to initialise WC_Filesystem API', 'woocommerce' ) );
```

Comment:
Thanks! I wonder if we could tweak to provide some more context and also upgrade to warning level:

```suggestion
			wc_get_logger()->warning( __( 'Failed to initialise WC_Filesystem API while trying to update the MaxMind Geolocation database.', 'woocommerce' ) );
```

We also wondered if it could be useful to add an admin notice; however, we don't need to do that here as this is already a big improvement.